tap/extensions/http/http2_assembler.go

```diff
@@ -115,22 +138,50 @@ func (ga *GrpcAssembler) readMessage() (uint32, interface{}, error) {
 	dataString := base64.StdEncoding.EncodeToString(data)
 
 	// Use http1 types only because they are expected in http_matcher.
-	// TODO: Create an interface that will be used by http_matcher:registerRequest and http_matcher:registerRequest
-	//       to accept both HTTP/1.x and HTTP/2 requests and responses
-	var messageHTTP1 interface{}
-	if _, ok := headersHTTP1[":method"]; ok {
+	method := headersHTTP1.Get(":method")
+	status := headersHTTP1.Get(":status")
+
+	// gRPC detection
+	grpcStatus := headersHTTP1.Get("Grpc-Status")
+	if grpcStatus != "" {
+		isGrpc = true
+		status = grpcStatus
+	}
+
+	if strings.Contains(headersHTTP1.Get("Content-Type"), "application/grpc") {
+		isGrpc = true
+		grpcPath := headersHTTP1.Get(":path")
+		pathSegments := strings.Split(grpcPath, "/")
+		if len(pathSegments) > 0 {
+			method = pathSegments[len(pathSegments)-1]
+		}
+
+		// Split method by uppercase
+		re := regexp.MustCompile(`[A-Z][^A-Z]*`)
+		method = strings.Join(re.FindAllString(method, -1), " ")
```

Comment:
Why not leave the method as it is?
It's what a user knows from the gRPC stubs.

Reply:
Overriding "method" might be confusing. The HTTP/2 method is still POST.
(I saw in the images that you attached that there is a difference between method in headers, which is still POST, and method in details, which is the gRPC method. This is good).
I suggest that the gRPC method be kept in a separate field. Is this possible? Maybe if "path" is parsed later.

Reply:
Reverted the regex split with https://github.com/up9inc/mizu/commit/6fce0ff1fa6b0ae82198dd369d1d12d84d3ed021

It was there because in the UI they were capitalized.

Reply:
Yes in HTTP/2 the method is still `POST` but in terms of gRPC, the `:status` and `:method` pseudo-headers of HTTP/2 are meaningless. That's why we override the HTTP/2 status and method with gRPC status and method in case of gRPC.